integr8ly/application-monitoring-operator
# Allow blackbox targets to be specified via a BlackboxTargets CRD/CR INTLY-2618

## JIRA: https://issues.jboss.org/browse/INTLY-2618
Allow blackbox targets to be specified via a `BlackboxTargets` CRD/CR
Installer version bump PR: https://github.com/integr8ly/installation/pull/751

## Verification:
- Checkout code from my fork/branch ``davidkirwan:blackboxtargets_crd``
- Deploy, crd, and apply roles:
-- ``oc apply -f deploy/crds/BlackboxTarget.yaml``
-- ``oc apply -f deploy/operator_roles``
-- ``oc apply -f deploy/roles``
- If verifying this in an integreatly cluster which already has the Application Monitoring Operator running do the following:
-- In the namespace ``middleware-monitoring`` spool down the ``application monitoring operator`` instance to ``0``
- Launch the operator locally: ``operator-sdk up local --namespace "middleware-monitoring"``
- Create a BlackboxTarget CR: ``oc apply -f deploy/example/BlackboxTarget.yaml``
- Operator logs should show the new CR BlackboxTarget being reconciled
- Check the secret: ``additional-scrape-configs`` should have the ``targets`` array updated with the blackbox targets added via the BlackboxTarget CR
- Grafana dashboards should eventually pick these up also in the ``endpoints`` dashboards
